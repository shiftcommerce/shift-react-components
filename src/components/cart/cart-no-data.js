// Libraries
import React from 'react'
import classNames from 'classnames'

/**
 * Renders a help message when there is no data for the cart
 * @param  {Object} props
 * @return {string} - HTML markup for the component
 */
const CartNoData = (props) => {
  return (
    <section className={classNames(props.className, 'c-cart-no-data')}>
      <p>
        You have no items in your cart. If you are expecting items in it, please refresh.
      </p>
      <p>
        If issue still persists, please check with our customer care.
      </p>
    </section>
  )
}

export default CartNoData
