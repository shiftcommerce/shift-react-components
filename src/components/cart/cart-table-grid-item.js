// Libraries
import React from 'react'
import classNames from 'classnames'

/**
 * Renders the cart table grid item wrapper
 * @param  {Object} props
 * @return {string} - HTML markup for the component
 */
const CartTableGridItem = (props) => {
  return (
    <div className={classNames(props.className, 'c-cart-table__grid-item', { [`c-cart-table__grid-item--${props.item}`]: props.item })}>
      { props.children }
    </div>
  )
}

export default CartTableGridItem
