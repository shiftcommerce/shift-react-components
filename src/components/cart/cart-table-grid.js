// Libraries
import React from 'react'
import classNames from 'classnames'

/**
 * Renders the cart table grid wrapper
 * @param  {Object} props
 * @return {string} - HTML markup for the component
 */
const CartTableGrid = (props) => {
  return (
    <section className={classNames(props.className, 'c-cart-table__grid')}>
      { props.children }
    </section>
  )
}

export default CartTableGrid
