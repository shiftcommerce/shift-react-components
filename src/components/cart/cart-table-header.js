// Libraries
import React, { PureComponent } from 'react'
import classNames from 'classnames'
import Pluralize from 'react-pluralize'
import t from 'typy'
import format from 'date-fns/format'

// Lib
import businessDaysFromNow from '../../lib/business-days-from-now'

class CartTableHeader extends PureComponent {
  /**
   * Render the details of the basket, such as how many items are in the basket
   * @param  {Object} cart
   * @return {string} - HTML markup for the component
   */
  renderBasketDetails (cart) {
    return (
      <div className='c-cart-table__header-grid-item c-cart-table__header-grid-item--a'>
        <h1 className='c-cart-table__title'>Your Shopping Basket <a className='c-cart-table__amount'>({ cart.line_items_count || 0 })</a></h1>
        <div className='c-cart-table__text'>
          <p className='c-cart-table__description'>You have <a>{ cart.line_items_count || 0 }</a> <Pluralize singular='item' count={cart.line_items_count || 0} showCount={false} /> in your shopping basket.</p>
          <p className='c-cart-table__description'><Pluralize singular='This' plural='These' count={cart.line_items_count || 0} showCount={false} /> <Pluralize singular='item' count={cart.line_items_count || 0} showCount={false} /> will be saved for 48 hours depending on availability.</p>
        </div>
      </div>
    )
  }

  /**
   * Render the estimated delivery date
   * @param  {array}  shippingMethods
   * @return {string} - HTML markup for the component
   * @todo update to use actual delivery date, currently using data from fixture
   */
  renderDeliveryEstimate (shippingMethod) {
    const workingDays = t(shippingMethod, 'meta_attributes.working_days.value').safeObject

    return (
      <div className='c-cart-table__header-grid-item c-cart-table__header-grid-item--b'>
        <h1 className='c-cart-table__title'>Estimated Delivery</h1>
        <p className='c-cart-table__description'>We will deliver your item: { format(businessDaysFromNow(workingDays), 'dddd Do MMMM') }.</p>
      </div>
    )
  }

  render () {
    const { cart, className, shippingMethod } = this.props

    return (
      <section className={classNames(className, 'c-cart-table__header')}>
        { this.props.breadcrumb }
        <div className='c-cart-table__header-grid'>
          { this.renderBasketDetails(cart) }
          { this.renderDeliveryEstimate(shippingMethod) }
        </div>
      </section>
    )
  }
}

export default CartTableHeader
