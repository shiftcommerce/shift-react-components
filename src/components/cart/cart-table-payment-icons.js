// Libraries
import React from 'react'
import classNames from 'classnames'

// Lib
import componentMapping from '../../lib/component-mapping'

/**
 * Renders the payment icons
 * @param  {Object} props
 * @return {string} - HTML markup for the component
 */
const CartTablePaymentIcons = (props) => {
  const PaymentIcons = componentMapping('PaymentIcons')

  return (
    <div className={classNames(props.className, 'c-cart-table__icons')}>
      <PaymentIcons />
    </div>
  )
}

export default CartTablePaymentIcons
