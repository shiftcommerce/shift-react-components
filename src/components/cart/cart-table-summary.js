// Libraries
import React, { PureComponent } from 'react'
import classNames from 'classnames'

// Libs
import componentMapping from '../../lib/component-mapping'
import { decimalPrice } from '../../lib/decimal-price'

class CartTableSummary extends PureComponent {
  constructor (props) {
    super(props)

    this.Link = componentMapping('Link')
  }

  /**
   * Renders the promotions
   * @param  {Object}  cart
   * @return {string} - HTML markup for the component
   */
  renderPromotions (cart) {
    return cart.discount_summaries.map(discountSummary => {
      return (
        <dl className='c-cart-summary__promotion' key={ discountSummary.id }>
          <dt>{ discountSummary.name }</dt>
          <dd>- &pound;{ decimalPrice(discountSummary.total) }</dd>
        </dl>
      )
    })
  }

  /**
   * Renders the shipping price
   * @param  {Object}  cart
   * @param  {boolean} loading
   * @return {string} - HTML markup for the component
   */
  renderShippingPrice (cart, loading, cheapestShipping) {
    if (loading) {
      return <a>Estimating shipping cost...</a>
    } else if (cart.shipping_method) {
      return <a>{ `Shipping: £${decimalPrice(cart.shipping_method.total)}` }</a>
    } else {
      return <a>{ `Shipping from: £${cheapestShipping.total}` }</a>
    }
  }

  /**
   * Renders the cart total
   * @param  {Object}  cart
   * @param  {Object}  cheapestShipping
   * @return {string} - HTML markup for the component
   */
  renderTotal (cart, cheapestShipping) {
    if (cart.shipping_method) {
      return cart.total
    } else {
      return cart.total + cheapestShipping.total
    }
  }

  render () {
    const { cart, className, loading, cheapestShipping } = this.props

    return (
      <>
        <section className={classNames(className, 'c-cart-summary')}>
          <dl>
            <dt><a>Total Price:</a></dt>
            <dd><a>£{ decimalPrice(cart.sub_total) }</a></dd>
          </dl>
          { this.renderPromotions(cart) }
          <dl>
            <dt>{ this.renderShippingPrice(cart, loading, cheapestShipping) }</dt>
          </dl>
          <dl>
            <dt><p className='u-bold'>TOTAL:</p></dt>
            <dd><p className='u-bold'>£{ !loading && decimalPrice(this.renderTotal(cart, cheapestShipping)) }</p></dd>
          </dl>
          <dl>
            <a>* Including VAT</a>
          </dl>
        </section>
        <section className={classNames(className, 'c-cart-summary-buttons')}>
          <this.Link href='/slug?slug=/homepage' as='/' className='o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue'>
            Continue Shopping
          </this.Link>
          <this.Link href='/checkout/shipping-address' className='o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed'>
            Checkout
          </this.Link>
        </section>
      </>
    )
  }
}

export default CartTableSummary
