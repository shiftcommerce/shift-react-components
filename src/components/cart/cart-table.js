// Libraries
import React from 'react'
import classNames from 'classnames'

/**
 * Renders the cart table wrapper, and its respective child elements
 * @param  {Object} props
 * @return {string} - HTML markup for the component
 */
const CartTable = (props) => {
  return (
    <section className={classNames(props.className, 'c-cart-table')}>
      { props.children }
    </section>
  )
}

export default CartTable
