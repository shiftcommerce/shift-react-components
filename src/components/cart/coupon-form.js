// Libraries
import React, { Component } from 'react'
import { Formik, Form, Field, ErrorMessage } from 'formik'
import PropTypes from 'prop-types'

// Lib
import componentMapping from '../../lib/component-mapping'

class CouponForm extends Component {
  constructor (props) {
    super(props)

    this.Button = componentMapping('Button')
  }

  /**
   * Renders a vailation message if no coupon code is entered
   * @param  {Object} values
   * @return {string} - HTML markup for the component
   */
  validate (values) {
    let errors = {}
    if (!values.couponCode) errors.couponCode = 'Please enter a coupon code'
    return errors
  }

  /**
   * Renders the form
   * @param  {function} validate
   * @param  {function} handleSubmit
   * @return {string} - HTML markup for the component
   */
  renderForm () {
    return (
      <Formik
        initialValues={{ couponCode: '' }}
        validate={this.validate}
        onSubmit={this.props.handleSubmit}
      >
        { () => (
          <Form>
            <div className='c-coupon-form__field'>
              <Field type='text' name='couponCode' className='c-coupon-form__input' placeholder='Enter Promotion Code' />
              <div className="c-coupon-form__button-container">
                <this.Button
                  aria-label='Apply Gift Code'
                  className='c-coupon-form__button o-button--sml'
                  label='Apply'
                  status='primary' />
              </div>
            </div>
            <ErrorMessage name='couponCode' component='div' className='c-coupon-form__error' />
          </Form>
        ) }
      </Formik>
    )
  }

  render () {
    return (
      <div className='c-coupon-form'>
        <div aria-label='Use gift card or rewards code' className='c-gift__wrapper'>
          <p className='c-coupon-form__title'>Promotion Code</p>
          { this.renderForm() }
        </div>
      </div>
    )
  }
}

CouponForm.propTypes = {
  handleSubmit: PropTypes.func
}

export default CouponForm
