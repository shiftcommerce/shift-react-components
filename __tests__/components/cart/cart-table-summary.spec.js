// Libraries
import React from 'react'

// Components
import CartTableSummary from '../../../src/components/cart/cart-table-summary'

// Fixtures
import cart from '../../fixtures/cart'

test('renders correct messages when cart has items', () => {
  const className = 'dummy-classname'

  // act
  const wrapper = mount(
    <CartTableSummary
      className={className}
      cart={cart}
      loading={false}
    />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£1.50 off all orders') // Discount
  expect(wrapper).toIncludeText('Shipping: £3.45') // Shipping total
  expect(wrapper.find('section')).toHaveClassName(className)
})

test('renders correct messages when loading shipping methods', () => {
  // act
  const wrapper = mount(
    <CartTableSummary
      cart={cart}
      loading={true}
    />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Estimating shipping cost...')
})

test('renders correct messages when cart has no shipping method', () => {
  const dummyShippingCost = 5
  const emptyCart = {
    line_items: [],
    sub_total: 10,
    line_items_count: 2,
    discount_summaries: [],
    total: 10
  }

  const cheapestShipping = {
    total: dummyShippingCost
  }

  // act
  const wrapper = mount(
    <CartTableSummary
      cart={emptyCart}
      loading={false}
      cheapestShipping={cheapestShipping}
    />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(`Shipping from: £${dummyShippingCost}`)
  expect(wrapper).toIncludeText('TOTAL:£15.00') // dummyShippingCost + emptyCart.total
})
