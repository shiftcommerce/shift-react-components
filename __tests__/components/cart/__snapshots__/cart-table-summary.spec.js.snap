// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct messages when cart has items 1`] = `
<CartTableSummary
  cart={
    Object {
      "channel": "web",
      "created_at": "2019-01-07T14:28:02Z",
      "discount_summaries": Array [
        Object {
          "id": "2918ed5e-2515-4603-b22f-aa1487304f05",
          "name": "£1.50 off all orders",
          "promotion_id": 21,
          "total": 1.5,
        },
      ],
      "email": "",
      "free_shipping": false,
      "id": 151,
      "line_items": Array [
        Object {
          "cart_id": 151,
          "container_id": 151,
          "gift": false,
          "id": "311",
          "item": Object {
            "available_to_browse": true,
            "available_to_order": true,
            "created_at": "2018-09-24T15:43:00Z",
            "current_price": 6.77,
            "deleted_at": null,
            "description": "Product 40 - variant 1 Grown at an altitude of 4, 500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them.They are picked by hand and sun- dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
            "ewis_eligible": false,
            "has_assets": true,
            "id": "321",
            "meta_attributes": Object {},
            "ordered_total": 10,
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "price": 6.77,
            "price_includes_taxes": true,
            "product": Object {
              "attribute_merchandising": Object {},
              "available_to_browse": true,
              "average_rating": 0,
              "canonical_path": "/jugs/seed_product_40",
              "category_ids_cache": Array [
                57,
              ],
              "content_type": "html",
              "created_at": "2018-09-24T15:43:00Z",
              "description": "Grown at an altitude of 4,500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them. They are picked by hand and sun-dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
              "ewis_status": "not_ewis_eligible",
              "has_assets": true,
              "id": "164",
              "max_current_price": 90.31,
              "max_price": 90.31,
              "merchandised": true,
              "merged_attribute_merchandising": Object {},
              "meta_attributes": Object {},
              "min_current_price": 6.77,
              "min_price": 6.77,
              "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
              "published": true,
              "rating": 0,
              "reference": "seed_product_40",
              "sellable": true,
              "seo_meta_description": "Product 40 - Qui sed maiores occaecati.",
              "seo_meta_keywords": "minus, ut, ut, adipisci, consequatur",
              "seo_meta_title": "SCS-04-CJ coffee jug 600ml",
              "slug": "jugs/seed_product_40",
              "title": "SCS-04-CJ coffee jug 600ml",
              "updated_at": "2018-11-21T10:16:59Z",
              "web_product_description": "Product 40 - Totam aut quo facere.",
              "web_product_name": "SCS-04-CJ coffee jug 600ml",
            },
            "product_id": 164,
            "reference": "est_36_1",
            "sku": "3008557600817",
            "stock_allocated_level": 15,
            "stock_available_level": 85,
            "stock_level": 100,
            "tax_code": "vat",
            "title": "Variant 1",
            "updated_at": "2018-11-20T13:18:46Z",
          },
          "item_id": 321,
          "item_type": "Variant",
          "line_item_discounts": Array [
            Object {
              "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
              "line_item_number": 1,
              "promotion_id": 21,
              "total": 1.5,
            },
          ],
          "sku": "3008557600817",
          "stock_available_level": 85,
          "sub_total": 6.77,
          "tax": 0.88,
          "tax_rate": "0.2",
          "title": "Variant 1",
          "total": 5.27,
          "total_discount": 1.5,
          "unit_price": 6.77,
          "unit_quantity": 1,
        },
      ],
      "line_items_count": 2,
      "meta_attributes": Object {},
      "shipping_method": Object {
        "total": 3.45,
      },
      "shipping_total": 3.45,
      "shipping_total_discount": 0,
      "sub_total": 6.77,
      "tax": 0.88,
      "test": false,
      "total": 5.27,
      "total_discount": 1.5,
      "updated_at": "2019-01-07T14:32:05Z",
    }
  }
  className="dummy-classname"
  loading={false}
>
  <section
    className="dummy-classname c-cart-summary"
  >
    <dl>
      <dt>
        <a>
          Total Price:
        </a>
      </dt>
      <dd>
        <a>
          £
          6.77
        </a>
      </dd>
    </dl>
    <dl
      className="c-cart-summary__promotion"
      key="2918ed5e-2515-4603-b22f-aa1487304f05"
    >
      <dt>
        £1.50 off all orders
      </dt>
      <dd>
        - £
        1.50
      </dd>
    </dl>
    <dl>
      <dt>
        <a>
          Shipping: £3.45
        </a>
      </dt>
    </dl>
    <dl>
      <dt>
        <p
          className="u-bold"
        >
          TOTAL:
        </p>
      </dt>
      <dd>
        <p
          className="u-bold"
        >
          £
          5.27
        </p>
      </dd>
    </dl>
    <dl>
      <a>
        * Including VAT
      </a>
    </dl>
  </section>
  <section
    className="dummy-classname c-cart-summary-buttons"
  >
    <Link
      as="/"
      className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue"
      href="/slug?slug=/homepage"
    >
      <a
        as="/"
        className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue"
        href="/slug?slug=/homepage"
      >
        Continue Shopping
      </a>
    </Link>
    <Link
      className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed"
      href="/checkout/payment-method"
    >
      <a
        className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed"
        href="/checkout/payment-method"
      >
        Checkout
      </a>
    </Link>
  </section>
</CartTableSummary>
`;

exports[`renders correct messages when cart has no shipping method 1`] = `
<CartTableSummary
  cart={
    Object {
      "discount_summaries": Array [],
      "line_items": Array [],
      "line_items_count": 2,
      "sub_total": 10,
      "total": 10,
    }
  }
  cheapestShipping={
    Object {
      "total": 5,
    }
  }
  loading={false}
>
  <section
    className="c-cart-summary"
  >
    <dl>
      <dt>
        <a>
          Total Price:
        </a>
      </dt>
      <dd>
        <a>
          £
          10.00
        </a>
      </dd>
    </dl>
    <dl>
      <dt>
        <a>
          Shipping from: £5
        </a>
      </dt>
    </dl>
    <dl>
      <dt>
        <p
          className="u-bold"
        >
          TOTAL:
        </p>
      </dt>
      <dd>
        <p
          className="u-bold"
        >
          £
          15.00
        </p>
      </dd>
    </dl>
    <dl>
      <a>
        * Including VAT
      </a>
    </dl>
  </section>
  <section
    className="c-cart-summary-buttons"
  >
    <Link
      as="/"
      className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue"
      href="/slug?slug=/homepage"
    >
      <a
        as="/"
        className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue"
        href="/slug?slug=/homepage"
      >
        Continue Shopping
      </a>
    </Link>
    <Link
      className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed"
      href="/checkout/payment-method"
    >
      <a
        className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed"
        href="/checkout/payment-method"
      >
        Checkout
      </a>
    </Link>
  </section>
</CartTableSummary>
`;

exports[`renders correct messages when loading shipping methods 1`] = `
<CartTableSummary
  cart={
    Object {
      "channel": "web",
      "created_at": "2019-01-07T14:28:02Z",
      "discount_summaries": Array [
        Object {
          "id": "2918ed5e-2515-4603-b22f-aa1487304f05",
          "name": "£1.50 off all orders",
          "promotion_id": 21,
          "total": 1.5,
        },
      ],
      "email": "",
      "free_shipping": false,
      "id": 151,
      "line_items": Array [
        Object {
          "cart_id": 151,
          "container_id": 151,
          "gift": false,
          "id": "311",
          "item": Object {
            "available_to_browse": true,
            "available_to_order": true,
            "created_at": "2018-09-24T15:43:00Z",
            "current_price": 6.77,
            "deleted_at": null,
            "description": "Product 40 - variant 1 Grown at an altitude of 4, 500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them.They are picked by hand and sun- dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
            "ewis_eligible": false,
            "has_assets": true,
            "id": "321",
            "meta_attributes": Object {},
            "ordered_total": 10,
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
            "price": 6.77,
            "price_includes_taxes": true,
            "product": Object {
              "attribute_merchandising": Object {},
              "available_to_browse": true,
              "average_rating": 0,
              "canonical_path": "/jugs/seed_product_40",
              "category_ids_cache": Array [
                57,
              ],
              "content_type": "html",
              "created_at": "2018-09-24T15:43:00Z",
              "description": "Grown at an altitude of 4,500 feet, the tea bush 'wakes' after the winter in April and produces delicate white buds, which often have a white down on them. They are picked by hand and sun-dried so that fermentation cannot take place. Very little white tea is produced, as this process requires particular care, increasing its rarity and making it highly sought after by tea buyers all over the world. It has a delicate flavour, so is best drunk without milk.",
              "ewis_status": "not_ewis_eligible",
              "has_assets": true,
              "id": "164",
              "max_current_price": 90.31,
              "max_price": 90.31,
              "merchandised": true,
              "merged_attribute_merchandising": Object {},
              "meta_attributes": Object {},
              "min_current_price": 6.77,
              "min_price": 6.77,
              "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/131/1537803713.5364385-SCS-04-CJ_coffee_jug_600ml.jpg",
              "published": true,
              "rating": 0,
              "reference": "seed_product_40",
              "sellable": true,
              "seo_meta_description": "Product 40 - Qui sed maiores occaecati.",
              "seo_meta_keywords": "minus, ut, ut, adipisci, consequatur",
              "seo_meta_title": "SCS-04-CJ coffee jug 600ml",
              "slug": "jugs/seed_product_40",
              "title": "SCS-04-CJ coffee jug 600ml",
              "updated_at": "2018-11-21T10:16:59Z",
              "web_product_description": "Product 40 - Totam aut quo facere.",
              "web_product_name": "SCS-04-CJ coffee jug 600ml",
            },
            "product_id": 164,
            "reference": "est_36_1",
            "sku": "3008557600817",
            "stock_allocated_level": 15,
            "stock_available_level": 85,
            "stock_level": 100,
            "tax_code": "vat",
            "title": "Variant 1",
            "updated_at": "2018-11-20T13:18:46Z",
          },
          "item_id": 321,
          "item_type": "Variant",
          "line_item_discounts": Array [
            Object {
              "id": "2a720d8d-ae39-462e-b2e2-4c49ca5c8846",
              "line_item_number": 1,
              "promotion_id": 21,
              "total": 1.5,
            },
          ],
          "sku": "3008557600817",
          "stock_available_level": 85,
          "sub_total": 6.77,
          "tax": 0.88,
          "tax_rate": "0.2",
          "title": "Variant 1",
          "total": 5.27,
          "total_discount": 1.5,
          "unit_price": 6.77,
          "unit_quantity": 1,
        },
      ],
      "line_items_count": 2,
      "meta_attributes": Object {},
      "shipping_method": Object {
        "total": 3.45,
      },
      "shipping_total": 3.45,
      "shipping_total_discount": 0,
      "sub_total": 6.77,
      "tax": 0.88,
      "test": false,
      "total": 5.27,
      "total_discount": 1.5,
      "updated_at": "2019-01-07T14:32:05Z",
    }
  }
  loading={true}
>
  <section
    className="c-cart-summary"
  >
    <dl>
      <dt>
        <a>
          Total Price:
        </a>
      </dt>
      <dd>
        <a>
          £
          6.77
        </a>
      </dd>
    </dl>
    <dl
      className="c-cart-summary__promotion"
      key="2918ed5e-2515-4603-b22f-aa1487304f05"
    >
      <dt>
        £1.50 off all orders
      </dt>
      <dd>
        - £
        1.50
      </dd>
    </dl>
    <dl>
      <dt>
        <a>
          Estimating shipping cost...
        </a>
      </dt>
    </dl>
    <dl>
      <dt>
        <p
          className="u-bold"
        >
          TOTAL:
        </p>
      </dt>
      <dd>
        <p
          className="u-bold"
        >
          £
        </p>
      </dd>
    </dl>
    <dl>
      <a>
        * Including VAT
      </a>
    </dl>
  </section>
  <section
    className="c-cart-summary-buttons"
  >
    <Link
      as="/"
      className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue"
      href="/slug?slug=/homepage"
    >
      <a
        as="/"
        className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--continue"
        href="/slug?slug=/homepage"
      >
        Continue Shopping
      </a>
    </Link>
    <Link
      className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed"
      href="/checkout/payment-method"
    >
      <a
        className="o-button--sml c-cart-summary-buttons__cta c-cart-summary-buttons__cta--proceed"
        href="/checkout/payment-method"
      >
        Checkout
      </a>
    </Link>
  </section>
</CartTableSummary>
`;
